its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java

```diff
@@ -652,20 +653,26 @@ void testEsprojVueWithBackend() throws IOException {
     assertTrue(result.isSuccess());
 
     List<Issue> issues = TestUtils.allIssues(ORCHESTRATOR);
-    List<String> ruleKeys = issues.stream().map(Issue::getRule).collect(Collectors.toList());
-    assertThat(ruleKeys).hasSize(83);
-    assertThat(ruleKeys).contains(
-      SONAR_RULES_PREFIX + "S4487",
-      SONAR_RULES_PREFIX + "S1134",
-      "css:S4666",
-      "javascript:S2703",
-      "javascript:S2703",
-      "typescript:S3626");
-
-    // xml plugin results in 1 extra file and more lines
-    assertThat(TestUtils.getMeasureAsInteger(localProjectKey, "lines", ORCHESTRATOR)).isEqualTo(18681);
-    assertThat(TestUtils.getMeasureAsInteger(localProjectKey, "ncloc", ORCHESTRATOR)).isEqualTo(14028);
-    assertThat(TestUtils.getMeasureAsInteger(localProjectKey, "files", ORCHESTRATOR)).isEqualTo(213);
+    var version = ORCHESTRATOR.getServer().version();
+    var expectedIssues = new ArrayList<>(List.of(
+      tuple("csharpsquid:S1134", "my.project.14:AspBackend/Controllers/WeatherForecastController.cs"),
+      tuple("csharpsquid:S4487", "my.project.14:AspBackend/Controllers/WeatherForecastController.cs"),
+      tuple("typescript:S3626", "my.project.14:src/components/HelloWorld.vue"),
+      tuple("javascript:S2703", "my.project.14:src/main.js"),
+      tuple("javascript:S2703", "my.project.14:src/main.js")));
+    if (version.isGreaterThanOrEquals(2025, 1)) {
+      assertThat(issues)
+        .extracting(Issue::getRule, Issue::getComponent)
+        .containsExactlyInAnyOrder(expectedIssues.toArray(new Tuple[]{}));
+    } else {
+      assertThat(issues).hasSize(83);
+      assertThat(issues)
+        .extracting(Issue::getRule, Issue::getComponent)
+        .contains(expectedIssues.toArray(new Tuple[]{}));
+    }
+    assertThat(TestUtils.getMeasureAsInteger(localProjectKey, "lines", ORCHESTRATOR)).isBetween(300, 1300);
```

Comment:
Not really sure about this, but you may be able to use `isIn`:

```suggestion
    assertThat(TestUtils.getMeasureAsInteger(localProjectKey, "lines", ORCHESTRATOR)).isIn(300, 1300);
```

To have the exact value for the different versions of SQS.